### DISCUSSION TITLE
Rules for new lines before return and EOF

### DISCUSSION THREAD

**Discussion by itsjavi** [biomejs/biome#413] [Q&A] [answered]
1 - Is there any equivalent to eslint's `newline-before-return` rule? I cannot find it
2 - And also, is there a way for the formatter to always keep an empty line at the end of a file?

**Comment by ematipico** [comment, marked as answer]
> 1 - Is there any equivalent to eslint's `newline-before-return` rule? I cannot find it 

No there isn't.

> 2 - And also, is there a way for the formatter to always keep an empty line at the end of a file?

The formatter does that already. Check [the playground](https://biomejs.dev/playground/?indentStyle=space&quoteStyle=single&jsxQuoteStyle=single&lintRules=all&code=ZgB1AG4AYwB0AGkAbwBuACAAbgBhAG0AZQAoAHAAYQByAGEAbQBzACkAIAB7AAoAIAAgAAoAfQA%3D)

**Reply by itsjavi** [reply]
> `newline-before-return`

Would it be ok if I try to open a PR to add this rule?

**Reply by ematipico** [reply]
We tend to avoid formatting lint rules because we have the formatter for that.

**Reply by itsjavi** [reply]
so a javascript formatter option is better?
I would appreciate some guidance when it comes to propose these things :)

**Reply by StanislavVoronov** [reply]
Are there any activity on this rule?

**Reply by FranciscoKloganB** [reply]
Yes, please, a `newline-before-return` and `newline-before-conditional` and `newline-before-throw` would be amazing.

**Reply by robclancy** [reply]
"we don't lint that only format it" - isn't in the fucking formatter anyway and ignores questions about it
